docs/basics/101-126-config2.rst

```diff
@@ -0,0 +1,285 @@
+what should be annexed and what shouldn't be annexed up to individual file level.
+This can be very handy, and allows you to tune your dataset to your custom needs.
+For example, files you will often edit by hand could be stored in Git if they are
+not too large to ease modifying them [#f3]_.
+Once you know the basics of this type of configuration syntax, writing
+your own rules is easy. For more tips on how configure Git-annex's content
+management in ``.gitattributes``, take a look at `this <https://git-annex.branchable.com/tips/largefiles/>`_
+page of the Git-annex documentation.
+Later however you will see preconfigured DataLad *procedures* such as ``text2git`` that
+can apply useful configurations for you, just as ``text2git`` added the last line
+in the root ``.gitattributes`` file.
+
+``.gitmodules``
+^^^^^^^^^^^^^^^
+
+On last configuration file we want to look at is the ``.gitmodules`` file.
+There is one right in the root of your dataset:
+
+.. runrecord:: _examples/DL-101-125-108
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat .gitmodules
+
+Based on these contents, you might have already guessed what this file
+stores. ``.gitmodules`` is a configuration file that stores the mapping between
+your own dataset and any subdatasets you have installed in it.
+There will be an entry for each submodule (subdataset) in your dataset.
+The name *submodule* is Git terminology, and describes a Git repository inside of
+another Git repository, i.e. the super- and subdataset principles.
+Upon sharing your dataset, the information about subdatasets and where to retrieve
+them from is stored and shared with this file.
+
+
+
+``.datalad/config``
+^^^^^^^^^^^^^^^^^^^
+
+DataLad adds a repository-specific configuration file as well.
+It can be found in the ``.datalad`` directory, and just like ``.gitattributes``
+and ``.gitmodules`` it is version controlled and is thus shared together with
+the dataset. One can configure
+`many options <http://docs.datalad.org/en/latest/generated/datalad.config.html>`_,
+but currently, our ``.datalad/config`` file only stores a dataset ID:
+
+.. runrecord:: _examples/DL-101-125-106
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat .datalad/config
+
+
+Writing to configuration files other than ``.git/config``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+"Didn't you say that knowing git :command:`git config` command is already
+half of what I need to know?" you ask. "Now there are three other configuration
+files, and I don't know with which command I can write into these files."
+
+"Excellent question", you hear in return, "but in reality, you **do** know:
+it's also the :command:`git config` command. The only part of it you need to
+adjust is the ``-f``, ``--file`` parameter. By default, the command writes to
+a Git config file. But it can write to a different file if you specify it
+appropriately. For example
```

Comment:
Maybe reiterate that .gitattributes use a different format and cannot be edited in this way.